Cleaned up GA Flags

ronaldlangeveld opened the pull request:
ref https://github.com/TryGhost/Team/issues/2126

Cleaned up redundant GA feature flags.

- [x] There's a clear use-case for this code change, explained below
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test:all` and `yarn lint`)

ronaldlangeveld committed:
Removed newsletterPaywall from GA
ref https://github.com/TryGhost/Team/issues/2126

- removed usage of newsletterPaywall
- just commented out for now

ronaldlangeveld committed:
Removed `compExpiring` flag
ref https://github.com/TryGhost/Team/issues/2126

- removes `compExpiring` flag from GA features list.

ronaldlangeveld committed:
Removed fixNewsletterLinks flag
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Removed `freeTrial` labs GA flag
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Removed `emailAlerts` GA flags
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Cleaned up comments
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Fixed linting errors
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Merge branch 'main' into ga-flag-cleanup-2126

ronaldlangeveld committed:
fixed more linting errors
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Merge branch 'main' into ga-flag-cleanup-2126

daniellockyer commented on ghost/core/core/server/models/member.js:
```diff
@@ -154,11 +154,7 @@ const Member = ghostBookshelf.Model.extend({
                 // avoids bookshelf adding a `DISTINCT` to the query
                 // we know the result set will already be unique and DISTINCT hurts query performance
-                if (labs.isSet('compExpiring')) {
-                    qb.columns('products.*', 'expiry_at');
```
If we're removing `compExpiring`, do we not need to keep `                    qb.columns('products.*', 'expiry_at');`?

ronaldlangeveld committed:
Added `expiry_at` to products query.
ref https://github.com/TryGhost/Ghost/pull/15762 https://github.com/TryGhost/Ghost/pull/15762#discussion_r1012751191

- accidentally removed while getting rid of the `compExpiring` flag.

ronaldlangeveld committed:
Merge branch 'main' into ga-flag-cleanup-2126

ronaldlangeveld committed:
Updated snapshots
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Merge branch 'main' into ga-flag-cleanup-2126

kevinansfield commented on ghost/admin/app/components/modal-tier.hbs:
```diff
@@ -305,22 +250,18 @@
                                                 </span>
                                             {{/if}}
                                         {{/if}}
-                                    {{else}}
```
I think the removed `{{else}}` should be the one above? This one is the opposite of the `{{#if this.stripeYearlyAmount}}` line.

Sidenote: The improper indenting here makes the conditionals harder to follow 😓

kevinansfield commented on ghost/admin/app/templates/offer.hbs:
```diff
@@ -47,18 +47,14 @@
                             <span>Discount information</span>
                             <span class="midgrey gh-main-section-header-info tooltip-left" data-tooltip="Once saved, this cannot be changed" >{{svg-jar "info"}}</span>
                         </h4>
-                        {{else}}
                         <h4 class="gh-main-section-header small bn">
                             Discount information <span class="midgrey"> &ndash; Once saved, this cannot be changed.</span>
                         </h4>
-                        {{/if}}
```
This whole block needs removing, right now discount info is duplicated.

ronaldlangeveld committed:
Updated `.lint-todo`
ref https://github.com/TryGhost/Team/issues/2126

- false positive caused by the linter not ignore the table.
- ran `yarn lint:hbs --update-todo` to update the `.lint-todo` file which solves the linting error.

ronaldlangeveld committed:
Fixed linting errors
https://github.com/TryGhost/Team/issues/2126

- removed unused `labs` imports.

ronaldlangeveld committed:
Removed unneeded condition
ref https://github.com/TryGhost/Ghost/pull/15762#discussion_r1012876620

ronaldlangeveld committed:
Removed `and` in hbs condition.
ref https://github.com/TryGhost/Ghost/pull/15762#discussion_r1012878199

ronaldlangeveld committed:
Fixed conditions
ref https://github.com/TryGhost/Ghost/pull/15762#discussion_r1012880962

ronaldlangeveld committed:
Fixed duplicated discount info
ref https://github.com/TryGhost/Ghost/pull/15762#discussion_r1012887292

ronaldlangeveld committed:
Replaced `not` with unless.
ref https://github.com/TryGhost/Ghost/pull/15762#discussion_r1012888668

ronaldlangeveld committed:
Corrected modal-tier conditions
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Removed feature conditions
ref https://github.com/TryGhost/Team/issues/2126

ronaldlangeveld committed:
Ensure wrong `else` isn't removed
ref https://github.com/TryGhost/Ghost/pull/15762#discussion_r1012880962

ronaldlangeveld committed:
Merge branch 'main' into ga-flag-cleanup-2126

ronaldlangeveld committed:
Merge branch 'main' into ga-flag-cleanup-2126

ronaldlangeveld committed:
Fixed conditions logic for feature flags

ronaldlangeveld committed:
Fixed linting

ronaldlangeveld committed:
Merge branch 'main' into ga-flag-cleanup-2126

ronaldlangeveld committed:
Merge branch 'main' into ga-flag-cleanup-2126
